### DISCUSSION TITLE
Zwave-js drastically reduced transmit range of Aeotec Z-Stick ZW090

### DISCUSSION THREAD

**Discussion by TheOriginalMrWolf** [zwave-js/zwave-js#6502] [Request Support / Investigate Issue]
### Checklist

- [X] I have read and followed the above instructions
- [X] I have checked the [troubleshooting section](https://zwave-js.github.io/node-zwave-js/#/troubleshooting/index) and my problem is **not** described there.

- [X] I have read the [changelog](https://github.com/zwave-js/node-zwave-js/blob/master/CHANGELOG.md) and my problem was **not** mentioned there or the fix **did not work**.

### Describe the issue

Have just migrated from OpenHAB to Home Assistant (using zwave-js) after my OH server drive died and have run into a weird issue whereby it _appears_ that zwave-js dramatically reduces the transmit range of my Aeotec Z-Stick ZW090 (FW 1.2, SDK: v6.81.6).

I currently have 6 nodes on the network, the furthest being about 15m away (through some solid walls).

OH driving the Aeotec stick could talk directly to the furthest node, no problems.

After moving to zwave-js transmit range dropped dramatically.  I did some open-air line-of-sight tests and the furthest it can now reach is about 5m in clear air.  Server location, hardware, environment, Z-Stick, and connection are in identical configuration - I just replaced the dead drive & deployed HA/Zwave-js

I checked the Aeotec default power output setting (using the Aeotec backup tool) and it is still at full power.  I also did some tests changing the power setting from 0 - 10 in several steps with no perceivable difference.  Is zwave-js somehow telling the stick to reduce transmit power?

During testing it was interesting to note that, as the test node moved further away, the Aeotec stick still reported receiving messages from the node, but the node stopped responding to messages from the stick.

Temporary workaround was to put another ZWave node about 1m away to act as a repeater, but it's not ideal.  Also, interestingly, I don't seem to have the full complement of settings for the Z-Stick available through the UI.  For instance, can't turn the LED on/off (option not presented).

I thought I could up transmit power by changing settings but this doesn't make any difference.  Checking logs I see that attempts to change transmit power fail (and it also seems to struggle with soft reset).

Hope you can point me in the right direction please?

### Software versions

zwave-js-ui: 9.1.2.a13a720
zwave-js: 12.0.4

### Device information

Aeotec Z-Stick ZW090, node 1

### Checklist

- [X] I [made sure](https://zwave-js.github.io/node-zwave-js/#/troubleshooting/index?id=rule-1-use-the-driver-log-luke) to provide a **driver log** on level **debug**.

- [X] The log includes a re-interview of the problematic device _(if applicable)_.

- [X] The log includes the problematic interaction with the device _(if applicable)_.

- [X] I provided the node ID of the problematic device _(if applicable)_.

### Upload Logfile

[zwave-log.txt](https://github.com/zwave-js/node-zwave-js/files/13317980/zwave-log.txt)

[a comment by an automated account is not shown]

**Comment by AlCalzone** [comment]
Z-Wave JS doesn't change the powerlevel unless you instruct it to. Although a transmit option exists to send with low power, Z-Wave JS doesn't use that.

From your log it looks like the controller uses the lowest possible speed 9.6 kbps for most nodes, except Node 029, for which 100 kbps is used. This indicates there was a range issue before where the controller changed routes a few times until the low-speed one worked.
Not sure how exactly you performed your range tests, did you use priority routes to force a direct connection attempt?
In any case you may want to rebuild routes, hoping that the controller resorts to the faster routes again.

> I don't seem to have the full complement of settings for the Z-Stick available through the UI. For instance, can't turn the LED on/off (option not presented)

That uses custom Serial API commands which we haven't implemented.

**Comment by TheOriginalMrWolf** [comment]
Thanks for looking at this @AlCalzone!

>Z-Wave JS doesn't change the powerlevel unless you instruct it to
 
So, I configure Z-Wave JS to change the power level & this fails.  You can see this near the beginning of the logfile. Is there a break in the implementation?  Aeotec doc seems to say there is a single power level setting, basically a number 1-10, but the integration tries to use the set dBm interface? See:
```
19:21:32.141 CNTRLR   Querying configured powerlevel...
19:21:32.144 SERIAL » 0x0104000b08f8                                                       (6 bytes)
19:21:32.145 DRIVER » [REQ] [SerialAPISetup]
                        command: GetPowerlevel
                        payload: 0x08
19:21:32.147 SERIAL « [ACK]                                                                   (0x06)
19:21:32.149 SERIAL « 0x010b010b08000a00000a0000f6                                        (13 bytes)
19:21:32.150 SERIAL » [ACK]                                                                   (0x06)
19:21:32.151 DRIVER « [RES] [SerialAPISetup]
                        command:               GetPowerlevel
                        normal powerlevel:     0.0 dBm
                        output power at 0 dBm: 1.0 dBm
19:21:32.157 CNTRLR   Current powerlevel 0 dBm (1 dBm) differs from desired powerlevel 3 dBm (1 dBm)
                      , configuring it...
19:21:32.161 SERIAL » 0x0106000b041e0ae2                                                   (8 bytes)
19:21:32.162 DRIVER » [REQ] [SerialAPISetup]
                        command:               SetPowerlevel
                        normal powerlevel:     3.0 dBm
                        output power at 0 dBm: 1.0 dBm
19:21:32.164 SERIAL « [ACK]                                                                   (0x06)
19:21:32.165 SERIAL « 0x0105010b0400f4                                                     (7 bytes)
19:21:32.166 SERIAL » [ACK]                                                                   (0x06)
19:21:32.168 DRIVER « [RES] [SerialAPISetup]
                        command: SetPowerlevel
                        success: false
19:21:32.170 CNTRLR   The controller response indicated failure after 1/3 attempts. Scheduling next
                      try in 100 ms.
19:21:32.271 DRIVER » [REQ] [SerialAPISetup]
                        command:               SetPowerlevel
                        normal powerlevel:     3.0 dBm
                        output power at 0 dBm: 1.0 dBm
19:21:32.272 SERIAL » 0x0106000b041e0ae2                                                   (8 bytes)
19:21:32.275 SERIAL « [ACK]                                                                   (0x06)
19:21:32.276 SERIAL « 0x0105010b0400f4                                                     (7 bytes)
19:21:32.277 SERIAL » [ACK]                                                                   (0x06)
19:21:32.279 DRIVER « [RES] [SerialAPISetup]
                        command: SetPowerlevel
                        success: false
19:21:32.281 CNTRLR   The controller response indicated failure after 2/3 attempts. Scheduling next
                      try in 1100 ms.
```
 

> exactly you performed your range tests, did you use priority routes to force a direct connection attempt
 
Yes, I forced direct routes

> rebuild routes, hoping that the controller resorts to the faster routes again
 
Tried. Not much difference though unfortunately

**Reply by AlCalzone** [reply]
> Aeotec doc seems to say there is a single power level setting

Do they have a proprietary command for this like for the LED control? And do you know if OpenHAB was configuring that?

**Reply by TheOriginalMrWolf** [reply]
The way OH seems to support this @AlCalzone is by simply sending a zwave command to the node to update a parameter.  In the case of the Z-Stick you can see its power configuration from [line 47 in config file here](https://github.com/openhab/org.openhab.binding.zwave/blob/ec827d2210c41417ca3b6b39be78bc75b6ee08b7/src/main/resources/OH-INF/thing/aeon/zw090_0_0.xml#L47) - so parameter 220, 1 byte, default value 10

This is sent to the node from [line 588 handleConfigurationUpdate](https://github.com/openhab/org.openhab.binding.zwave/blob/ec827d2210c41417ca3b6b39be78bc75b6ee08b7/src/main/java/org/openhab/binding/zwave/handler/ZWaveThingHandler.java#L588C17-L588C42) - sending the value as a z-wave message [here](https://github.com/openhab/org.openhab.binding.zwave/blob/ec827d2210c41417ca3b6b39be78bc75b6ee08b7/src/main/java/org/openhab/binding/zwave/handler/ZWaveThingHandler.java#L695).

That seemed to work.. or, at least... _not_ reduce the power output...

**Comment by TheOriginalMrWolf** [comment]
Hi @AlCalzone - do you think my interpretation of what OH is doing is correct?  Just send the command to the controller the same way commands are sent to any other Z-Wave nodes?  Is this something that Z-Wave JS could do?

**Comment by AlCalzone** [comment]
The interesting part would be knowing if the parameter is always set to 10 on startup or not. If the controller defaults to something below 10 and OpenHAB always made sure it is 10, this could explain what you're seeing in Z-Wave JS.
Configuration parameters of the currently connected controller cannot be changed like the ones for end nodes though, as no data is sent via RF. This is done through proprietary controller-specific commands, which we'd have to specifically implement for this controller.

**Reply by AlCalzone** [reply]
Just as a sanity check, as you wrote the Aeotec tool still reads that it uses the highest power - could you set up a barebones OH instance and see if the range "comes back" there? If this was a more widespread issue, I'd expect to get more reports than one tbh.

**Comment by TheOriginalMrWolf** [comment]
> cannot be changed like the ones for end nodes though, as no data is sent via RF. This is done through proprietary controller-specific commands, which we'd have to specifically implement for this controller

What happens if you just invoke send-via-RF and simply set destination as the controller?

> Just as a sanity check, as you wrote the Aeotec tool still reads that it uses the highest power - could you set up a barebones OH instance and see if the range "comes back" there? If this was a more widespread issue, I'd expect to get more reports than one tbh.

Sorry @AlCalzone - don't have the resources to set that up right now 😒

**Reply by AlCalzone** [reply]
> What happens if you just invoke send-via-RF and simply set destination as the controller?

Nothing.
